### ISSUE TITLE
Land tool improved clearance checks

### ISSUE THREAD

**Issue by Broxzier** [OpenRCT2/OpenRCT2#2307]
Whenever raising a piece of land where there is a path or a track in the way, it prevent you from moving it up, even though the path/track is on a slope.

![example](http://i.imgur.com/ZFdqbxn.png)

(There was a path, but I removed some parts to show it more clearly)

**Comment by Christipher** [comment]
I can reproduce this bug in the original game... this should be declared an original bug (it's easy to reproduce - just open up extreme heights and try to modify land below a path slope heading towards the high end of the slope and you should know what I mean).

**Comment by HaasJona** [comment]
I don't really see a bug there and I don't think the game should automatically convert a "bridge" to a normal ground based path.

**Comment by Gymnasiast** [comment]
I do consider it a bug on the grounds that it doesn't raise the land because something is in the way, yet will let you build a path if you terraform first. That's not very consistent, and probably caused by a simplistic collision check (in the same way it does this with tracks).

**Comment by Christipher** [comment]
On flat land it can actually go from ground to bridge arbitrarily... and I think this is related to #2296

**Comment by Broxzier** [comment]
I agree that it is a bug. When the ground is sloped already you are able to lower it, but you can't revert it without removing the track first.

@Christipher Seems to be very related indeed.

**Comment by jensj12** [comment]
It is the same issue code-wise. The land tool only checks full cells while other tools check quarter-cells.
